### ISSUE TITLE
Crash associated with issue #2400

### ISSUE THREAD

**Issue by Dmorgen** [smith-chem-wisc/MetaMorpheus#2454]
Hi,
I've used the patched version you've sent me. It crashed during the quantification stage. 
please see the attached log.

Thanks!
David.

[results_171224.txt](https://github.com/user-attachments/files/18165126/results_171224.txt)

**Comment by trishorts** [comment]
were you doing tmt?

**Comment by Dmorgen** [comment]
No. did I mark that by any chance?

**Comment by trishorts** [comment]
first line of the error message reference line 209. 
![image](https://github.com/user-attachments/assets/e101e6b7-ec54-408c-966f-1d1a6b2a4694)
can you check the toml?

**Comment by Dmorgen** [comment]
No. But I will make sure the next attempt won’t have it. Thanks!

D.

**Comment by trishorts** [comment]
can you post the search toml for me

**Comment by trishorts** [comment]
i ran all your data through to quant on my side w/ no crash.

**Comment by Dmorgen** [comment]
Thanks! Will try again.

D.

**Comment by acesnik** [comment]
🤖 Thanks for posting this, David, and sorry it took a while to circle back!

Closing this one — we could never reproduce it (your data ran through to quant cleanly on our side, and we never got the search TOML to compare settings), and the crash you hit can no longer stop a run: #2574 wrapped quantification in crash handling, so a failure there now writes a `Quantification_crash.txt` and lets the rest of the search finish.

We did find the mechanism, and there's a fix in flight. `QuantificationAnalysis` dereferenced a nullable monoisotopic mass without checking it, and the filter ahead of it only screens on sequence — a resolved sequence doesn't imply a resolved mass. Two ways that happens: `Residue.ResidueMonoisotopicMass` is `NaN` for B, X and Z, and every NaN comparison is false, so even a completely unambiguous match came out null; or two matches whose modifications share a name but not a mass, which stringify to the same `FullSequence`. Either one used to cost quantification for the whole run rather than just the affected PSM. The fix drops those PSMs with a warning naming the count.

Your build's `PostSearchAnalysisTask.cs:209` was a red herring, incidentally — it was a custom patch, and line 209 in the tree at the time is a plain `bool` check, which is why the TMT question didn't lead anywhere.

Happy to reopen if you see anything like this on a current release. Thanks again for the detailed logs!
